frontend/src/app/university-search.service.ts

```diff
@@ -0,0 +1,34 @@
+import { Injectable } from '@angular/core';
+import { Http } from '@angular/http';
+
+import { Observable } from 'rxjs/Observable';
+import 'rxjs/add/operator/map';
+import 'rxjs/add/operator/toPromise';
+import 'rxjs/add/operator/share';
+import { UniversityResult } from './university-result';
+
+@Injectable()
+export class UniversitySearchService {
+    constructor(private http: Http) {}
+
+    search(term1: string, selectedRoute1: string, term2: string, selectedRoute2: string): Observable<UniversityResult[]> {
```

Comment:
You should use angular's HttpParams instead of manually appending to a string. You'd have to do things like space escaping manually. HttpParams will handle that for you.

Also selectedRoute1, term1, selectedRoute2, term2 are not understandable as names. They should be more descriptive.